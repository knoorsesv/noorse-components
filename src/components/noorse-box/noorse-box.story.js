import {console, document} from 'global'
import {storiesOf} from '@storybook/html'

storiesOf('Containers', module)
  .add('box', () => '<noorse-box>Content</noorse-box>')
  .add('box with header', () => '<noorse-box header="Header">Content</noorse-box>')
