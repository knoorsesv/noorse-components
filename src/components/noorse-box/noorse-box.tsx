import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'noorse-box',
  styleUrl: 'noorse-box.scss',
  shadow: true
})
export class NoorseBox {
  /**
   * The first name
   */
  @Prop() header: string;


  render() {
    return <div class="box">
      <h1>{this.header}</h1>
      <slot/>
    </div>;
  }
}
