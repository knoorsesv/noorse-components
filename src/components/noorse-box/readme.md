# noorse-box



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description    | Type     | Default     |
| -------- | --------- | -------------- | -------- | ----------- |
| `header` | `header`  | The first name | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
