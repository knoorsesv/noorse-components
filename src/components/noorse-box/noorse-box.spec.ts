import {newSpecPage} from "@stencil/core/testing";
import {NoorseBox} from "./noorse-box";
import {SpecPage} from "@stencil/core/dist/declarations";

describe("Noorse Box", function () {

  let box: SpecPage;

  beforeEach(async () => {
    box = await newSpecPage({
      components: [NoorseBox],
      html: `<noorse-box header="header">content</noorse-box>`
    });
    await box.waitForChanges()
  });

  it("should render a box with the content", async () => {
    expect(box.root.querySelector('div.box')).not.toBeNull()
  });

  it('should have a header tag with the input value', function () {
    expect(box.root.querySelector('h1').textContent).toContain('header')
  });

  it('should render the content inside of the div', function () {
    expect(box.root.querySelector('div.box').textContent).toContain('content')
  });

  it('should render correct snapshots', function () {
    expect(box.root.outerHTML).toMatchSnapshot()
  });

});
