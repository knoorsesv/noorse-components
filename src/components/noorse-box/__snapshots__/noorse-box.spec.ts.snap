// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Noorse Box should render correct snapshots 1`] = `"<noorse-box header=\\"header\\"><!----> <div class=\\"box\\"><h1>header</h1> content</div></noorse-box>"`;
